org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -249,6 +249,13 @@ public void didChangeConfiguration(DidChangeConfigurationParams params) {
 				unregisterCapability(Preferences.RENAME_ID, Preferences.TEXT_DOCUMENT_RENAME);
 			}
 		}
+		if (preferenceManager.getClientPreferences().isExecuteCommandDynamicRegistrationSupported()) {
+			if (preferenceManager.getPreferences().isExecuteCommandEnabled()) {
+				registerCapability(Preferences.EXECUTE_COMMAND_ID, Preferences.WORKSPACE_EXECUTE_COMMAND);
```

Comment:
pretty sure you need to return `new ExecuteCommandOptions(new ArrayList<>(commands))` (see CodeLensOptions L233)

Reply:
You'll need to update JDTLanguageServerTest as well